lib/src/Pos/Util/Servant.hs

```diff
@@ -988,3 +992,39 @@ single theData = WalletResponse {
     , wrStatus = SuccessStatus
     , wrMeta   = Metadata (PaginationMetadata 1 (Page 1) (PerPage 1) 1)
     }
+
+-- | Generates the @Content-Type: application/json@ 'HTTP.Header'.
+applicationJson :: HTTPTypes.Header
+applicationJson =
+    (hContentType, "application/json")
+
+-- | An error for representing unknown problems in the API. The Jsen
+newtype UnknownError = UnknownError Text
+    deriving (Show, Generic)
+
+deriveGeneric ''UnknownError
+
+instance Exception UnknownError
+
+instance HasDiagnostic UnknownError where
+    getDiagnosticKey (UnknownError msg) = msg
```

Comment:
The diagnostic key needs to be unique per constructor; We use to use `"msg"` for unknown errors in the wallet :+1:

Reply:
oh, well, then we should probably make this a type class with a bidirectional fundep with the second parameter being the `Symbol`! That'd enforce uniqueness :)

I guess I am confused on the `getDiagnosticKey` meaning. What does the class do wrt generating the Jsend error?